amr-wind/equation_systems/icns/source_terms/DragForcing.cpp

```diff
@@ -266,6 +267,35 @@ void DragForcing::operator()(
             bc_forcing_x = -(uxTarget - ux1) / dt;
             bc_forcing_y = -(uyTarget - uy1) / dt;
         }
+        const amrex::Real cell_drag = drag(i, j, k) / (drag(i, j, k) + tiny);
+        //! Cannot be used for waves
+        if (drag(i, j, k) > 1 && (!is_laminar) && (!is_waves)) {
+            const amrex::Real z0 = std::max(terrainz0(i, j, k), z0_min);
+            const amrex::GpuArray<amrex::Real, 4> cell_wind_x = {
```

Comment:
Ok sorry I had something simpler in mind. Create a GpuArray of Intvects
```
const amrex::IntVect ivip = iv + amrex::IntVect::TheDimensionVector(0);
const amrex::IntVect ivim = iv - amrex::IntVect::TheDimensionVector(0);
const amrex::IntVect ivjp = iv + amrex::IntVect::TheDimensionVector(1);
const amrex::IntVect ivjm = iv - amrex::IntVect::TheDimensionVector(1);
stuff into a gpuarray: iv_array (or init those straight up into that array)
loop over those (for const auto& iv_idx : iv_array) and call
compute_target_wind(
                        vel(iv_idx, 0), vel(iv_idx, 1),
                      etc);
bc_forcing_x +=
                    -(tmp_wind_target[0] - ux1) / dt * blank(iv_idx);
sum_blank += blank(iv_idx;
```

Should be more straitghtforward